diff -ur ../openldap-2.4.33/include/ldap_cdefs.h ./include/ldap_cdefs.h
--- ../openldap-2.4.33/include/ldap_cdefs.h	Wed Oct 10 05:18:49 2012
+++ ./include/ldap_cdefs.h	Mon Feb 10 04:50:33 2014
@@ -132,6 +132,9 @@
      (!defined(LDAP_LIBS_DYNAMIC) && defined(SLAPD_IMPORT)))
 #	define LBER_F(type)		extern __declspec(dllimport) type
 #	define LBER_V(type)		extern __declspec(dllimport) type
+#elif defined LBER_DLL_LIB
+#	define LBER_F(type)		__declspec(dllexport) type
+#	define LBER_V(type)		__declspec(dllexport) type
 #else
 #	define LBER_F(type)		extern type
 #	define LBER_V(type)		extern type
@@ -143,6 +146,9 @@
      (!defined(LDAP_LIBS_DYNAMIC) && defined(SLAPD_IMPORT)))
 #	define LDAP_F(type)		extern __declspec(dllimport) type
 #	define LDAP_V(type)		extern __declspec(dllimport) type
+#elif defined (LDAP_EXPORT)
+#	define LDAP_F(type)		__declspec(dllexport) type
+#	define LDAP_V(type)		__declspec(dllexport) type
 #else
 #	define LDAP_F(type)		extern type
 #	define LDAP_V(type)		extern type
diff -ur ../openldap-2.4.33/include/ldap_config.h ./include/ldap_config.h
--- ../openldap-2.4.33/include/ldap_config.h	Wed Oct 10 05:18:49 2012
+++ ./include/ldap_config.h	Mon Sep  9 23:45:50 2013
@@ -45,28 +45,28 @@
 
 /* directories */
 #ifndef LDAP_BINDIR
-#define LDAP_BINDIR			"%BINDIR%"
+#define LDAP_BINDIR			""
 #endif
 #ifndef LDAP_SBINDIR
-#define LDAP_SBINDIR		"%SBINDIR%"
+#define LDAP_SBINDIR		""
 #endif
 #ifndef LDAP_DATADIR
-#define LDAP_DATADIR		"%DATADIR%"
+#define LDAP_DATADIR		""
 #endif
 #ifndef LDAP_SYSCONFDIR
-#define LDAP_SYSCONFDIR		"%SYSCONFDIR%"
+#define LDAP_SYSCONFDIR		"c:\\openldap\\sysconf"
 #endif
 #ifndef LDAP_LIBEXECDIR
-#define LDAP_LIBEXECDIR		"%LIBEXECDIR%"
+#define LDAP_LIBEXECDIR		""
 #endif
 #ifndef LDAP_MODULEDIR
-#define LDAP_MODULEDIR		"%MODULEDIR%"
+#define LDAP_MODULEDIR         ""
 #endif
 #ifndef LDAP_RUNDIR
-#define LDAP_RUNDIR			"%RUNDIR%"
+#define LDAP_RUNDIR			""
 #endif
 #ifndef LDAP_LOCALEDIR
-#define LDAP_LOCALEDIR		"%LOCALEDIR%"
+#define LDAP_LOCALEDIR		""
 #endif
 
 
diff -ur ../openldap-2.4.33/libraries/liblber/nt_err.c ./libraries/liblber/nt_err.c
--- ../openldap-2.4.33/libraries/liblber/nt_err.c	Wed Oct 10 05:18:49 2012
+++ ./libraries/liblber/nt_err.c	Mon Feb 10 04:52:46 2014
@@ -23,7 +23,7 @@
 
 #define LBER_RETSTR( x ) case x: return #x;
 
-char *ber_pvt_wsa_err2string( int err )
+LBER_F( char * ) ber_pvt_wsa_err2string( int err )
 {
 	switch( err ) {
 		LBER_RETSTR( WSAEINTR )

--- ../openldap-2.4.44/include/ac/socket.h	2016-02-06 00:57:45.000000000 +0100
+++ ./include/ac/socket.h	2016-10-05 13:06:30.161290800 +0200
@@ -98,9 +98,15 @@
 #		define tcp_close( s )		closesocket( s )
 #	endif
 
+#ifndef EWOULDBLOCK
 #define EWOULDBLOCK WSAEWOULDBLOCK
+#endif
+#ifndef EINPROGRESS
 #define EINPROGRESS WSAEINPROGRESS
+#endif
+#ifndef ETIMEDOUT
 #define ETIMEDOUT	WSAETIMEDOUT
+#endif
 
 #undef	sock_errno
 #undef	sock_errstr

